File: GXpert/GXpert.Web/Modules/ServerTypes/Institute/BranchForm.ts
import { ServiceLookupEditor, LookupEditor, StringEditor, IntegerEditor, TextAreaEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface BranchForm {
    InstituteId: ServiceLookupEditor;
    DepartmentId: LookupEditor;
    Title: StringEditor;
    BranchCode: IntegerEditor;
    Description: TextAreaEditor;
}

export class BranchForm extends PrefixedContext {
    static readonly formKey = 'Institute.Branch';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!BranchForm.init)  {
            BranchForm.init = true;

            var w0 = ServiceLookupEditor;
            var w1 = LookupEditor;
            var w2 = StringEditor;
            var w3 = IntegerEditor;
            var w4 = TextAreaEditor;

            initFormType(BranchForm, [
                'InstituteId', w0,
                'DepartmentId', w1,
                'Title', w2,
                'BranchCode', w3,
                'Description', w4
            ]);
        }
    }
}

File: GXpert/GXpert.Web/Modules/Institute/Branch/BranchForm.cs
using Serenity.ComponentModel;
using System;

namespace GXpert.Institute.Forms;

[FormScript("Institute.Branch")]
[BasedOnRow(typeof(BranchRow), CheckNames = true)]
public class BranchForm
{
    public int InstituteId { get; set; }
    public int DepartmentId { get; set; }
    public string Title { get; set; }

    public int BranchCode { get; set; }
    [TextAreaEditor(Rows = 3)]
    public string Description { get; set; }
   
}